mlir/test/Conversion/TosaToMIOpen/tosa-to-miopen.mlir

```diff
@@ -0,0 +1,15 @@
+// RUN: mlir-opt --split-input-file --tosa-to-miopen-on-tensors %s -verify-diagnostics -o -| FileCheck %s
+
+// CHECK-LABEL: test_fusion
+// CHECK: miopen.conv2d(%{{.*}}, %{{.*}}, %{{.*}}) {arch = {{.*}}, dilations = [1 : i32, 1 : i32], filter_layout = ["k", "y", "x", "c", "g"], input_layout = ["ni", "hi", "wi", "ci", "gi"], num_cu = {{.*}} : i32, output_layout = ["no", "ho", "wo", "ko", "go"], padding = [0 : i32, 0 : i32, 0 : i32, 0 : i32], strides = [1 : i32, 1 : i32]} : memref<128x8x3x3x1xf32>, memref<128x8x32x32x1xf32>, memref<128x128x30x30x1xf32>
+    
+func @test_fusion(%arg0: tensor<128x8x32x32xf32>, %arg1: tensor<128x8x3x3xf32>, %arg2: tensor<8xf32>) -> tensor<128x128x30x30xf32> {
+  %0 = "tosa.conv2d"(%arg0, %arg1, %arg2) {dilation = [1, 1], pad = [0, 0, 0, 0], stride = [1, 1]} : (tensor<128x8x32x32xf32>, tensor<128x8x3x3xf32>, tensor<8xf32>) -> tensor<128x128x30x30xf32>
```

Comment:
[Question] Shall we need to provide client with this interface sometime in the future?

Reply:
Possibly if we want to optimize full or partial graph. Client could pass an IR in and we could pass an IR back with substitutions (fully realized kernels). For MIGraphX we could read ONNX directly and translate to TOSA. Alternatively we could have an API similar to NNAPI but which we stored internally as TOSA. Still fleshing out the ideas..

Reply:
Sounds good. I take your response as this being an area where a lot of change can happen. Therefore, if we want any optimization/update in the interface, we should hold off. Until we have defined the way client can take advantage for full-graph optimization, it is premature to decide on the implementation details of the specific interface.